### PR TITLE
Numerous bug fixes, ending in clean full LVS for both caravel and caravan.

### PR DESCRIPTION
**Pull request by RTimothyEdwards**
(1) Modified the .magicrc file to set a default for PDK if not set in
environment.  (2) Fixed the user ID programming layout to not leave holes
behind when the script moves the vias around (similar to the handling of
the GPIO defaults block).  (3) Added substrate isolation to gpio_control_block
and fixed the path references to the standard cells.  (4) Fixed the four
missing routes on the Caravan top level.  (5) Reinstated the large rendered
labels for the pads on both caravel and caravan.  (6) Corrected the top
level gate-level netlist for caravan to add the missing pins to the
management core wrapper.  (7) Did the same for the caravan top level RTL.
(8) Created scripts to run full LVS including extracting the management
core wrapper and reading all gate-level verilog submodules.  (9) Moved all
of the LVS scripts to the scripts directory.

### CODE REVIEW

**Comment by RTimothyEdwards** [comment]
@d-m-bailey :  I'd appreciate it if you could review these changes, especially around the user ID programming block.
Attn: @jeffdi

**Review by d-m-bailey — COMMENTED** [review]
Things you might want to take a look at.

1) In the `scripts/run_carav*_lvs_full.sh` files, the `readnet verilog` lines are all relative paths. Maybe `$CARAVEL_ROOT` and `$MCW_ROOT` would be better. I believe the management core is installed as `mcw_core_wrapper` by default, but the scripts reference `caravel_mgmt_soc_litex`.

2) The changes to `scripts/set_user_id.py` have not been included, so that will crash.

**Comment by d-m-bailey** [comment]
3 ) `simple_por` still uses a common ground. See #31

**Comment by RTimothyEdwards** [comment]
@jeffdi , @d-m-bailey :  I don't get why `scripts/set_user_id.py` is not changed.  That was in pull request #73, which was stated to have been merged 12 hours ago.  But I pulled the latest `main` branch and it's unchanged there.  What happened?

**Comment by RTimothyEdwards** [comment]
@jeffdi :  Did pull request #73 get blocked by failing the CI or something?  That's what's suggested in the Actions tab.  That's the only explanation I can come up with.  But I don't know why github claims that #73 was merged because clearly it wasn't.  It needs to be merged before this pull request can be merged.

**Comment by RTimothyEdwards** [comment]
@d-m-bailey :  I understand that the scripts are somewhat ad hoc, but my main goal was to get both projects to show LVS clean, and then move the scripts to the scripts/ directory so that they can be standardized and made more universally useful.  I can raise a separate issue to get them cleaned up, preferably with a hook to the Makefile so that one can do something like "make lvs-full" to initiate a complete LVS.

The simple POR fix is a separate issue, although since I can do the fix in probably an hour, max, it is well worth considering getting that done for this MPW.

**Comment by d-m-bailey** [comment]
@RTimothyEdwards Haven't completed LVS yet, but these are the temporary changes I made to `simple_por` (for reference).
simple_por_after

**Comment by d-m-bailey** [comment]
@RTimothyEdwards I had a problem trying to create a pull request where it didn't appear that the previous updates were included. Finally, this is what I did. `mpw5-set-user-id` is my local branch.
```
git checkout -b efabless-main mpw5-set-user-id
git pull https://github.com/efabless/caravel.git main
<fix conflicts>
git checkout mpw5-set-user-id
git merge --no-ff efabless-main
git push origin mpw5-set-user-id
```

**Comment by RTimothyEdwards** [comment]
@d-m-bailey : 
(1)  Add the comments about the simple_por to issue #31
(2) Please direct comments about merging to @jeffdi , as I do not have authority to do a merge myself.

**Comment by RTimothyEdwards** [comment]
@d-m-bailey :  Rather than try to get Jeff to work through the merge issues, I'm just going to make the changes from PR #73 in this PR so Jeff should only need to do the one merge.

**Comment by d-m-bailey** [comment]
@RTimothyEdwards the merge suggestions weren't for Jeff, but for your local repo. 
This is how I think it works.
You created the `all_lvs_clean` branch from `main`.
After that, the `main` github branch was updated and you pulled changes, but `all_lvs_clean` does not get updated it that case. I think you need to `git merge --no-ff`.

**Comment by RTimothyEdwards** [comment]
@d-m-bailey :  I am not well versed in any of the more obscure git options.  Anyway, I see your point, but on my laptop I had pulled the `main` branch and I was still not seeing the changes from your PR #73.  Anyway, I have merged those changes into this pull request, so they should get properly merged into `main` as soon as Jeff has the free time to do it.

**Review by jeffdi — APPROVED** [review]

**Comment by antonblanchard** [comment]
Hi @RTimothyEdwards, the removal of `-v $(PWD)/..:$(PWD)/..` broke tape outs for me, was that change intentional?

```
diff --git a/openlane/Makefile b/openlane/Makefile
index 8fd607d..ca6b7ed 100644
--- a/openlane/Makefile
+++ b/openlane/Makefile
@@ -44,8 +44,9 @@ endif
        @if [ -f ./$*/interactive.tcl ]; then\
                docker run --rm -v $(OPENLANE_ROOT):/openlane \
                -v $(PDK_ROOT):$(PDK_ROOT) \
-               -v $(PWD)/..:$(PWD)/.. \
+               -v $(MCW_ROOT):$(MCW_ROOT) \
                -v $(CARAVEL_ROOT):$(CARAVEL_ROOT) \
+               -e MCW_ROOT=$(MCW_ROOT) \
                -e PDK_ROOT=$(PDK_ROOT) \
                -e CARAVEL_ROOT=$(CARAVEL_ROOT) \
                -e PDK=$(PDK) \
@@ -56,8 +57,9 @@ endif
        else\
                docker run --rm -v $(OPENLANE_ROOT):/openlane \
                -v $(PDK_ROOT):$(PDK_ROOT) \
-               -v $(PWD)/..:$(PWD)/.. \
                -v $(CARAVEL_ROOT):$(CARAVEL_ROOT) \
+               -v $(MCW_ROOT):$(MCW_ROOT) \
+               -e MCW_ROOT=$(MCW_ROOT) \
                -e PDK=$(PDK) \
                -e PDK_ROOT=$(PDK_ROOT) \
                -e CARAVEL_ROOT=$(CARAVEL_ROOT) \
```

**Pull request merged**
